src/Workspaces/Remote/ServiceHub/Services/NavigateToSearch/RemoteNavigateToSearchService.cs

```diff
@@ -3,39 +3,43 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.NavigateTo;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Storage;
 using Roslyn.Utilities;
+using StreamJsonRpc;
 
 namespace Microsoft.CodeAnalysis.Remote
 {
     internal sealed class RemoteNavigateToSearchService : BrokeredServiceBase, IRemoteNavigateToSearchService
     {
-        internal sealed class Factory : FactoryBase<IRemoteNavigateToSearchService, IRemoteNavigateToSearchService.ICallback>
-        {
-            protected override IRemoteNavigateToSearchService CreateService(
-                in ServiceConstructionArguments arguments, RemoteCallback<IRemoteNavigateToSearchService.ICallback> callback)
-                => new RemoteNavigateToSearchService(arguments, callback);
-        }
+        /// <summary>
+        /// Navigate to is implemented using <see cref="IAsyncEnumerable{T}"/>.  This API works by having the client
+        /// "pull" for results from the server.  That means that the computation to produce the next result happens only
+        /// when the client is actually asking for that.  We would like to utilize resources more thoroughly when
+        /// searching, allowing for searching for results on multiple cores.  As such, we set a "max read ahead" amount
+        /// that allows the server to keep processing and producing results, even as the client is processing the batch
+        /// of results.
+        /// </summary>
+        /// <remarks>
+        /// This value was not determined empirically.
```

Comment:
Was there at least a hunch for the value? (but thank you for including this comment either way, since I hate magic constants that don't say they were picked out of thin air since you're terrified to touch them)